core/src/main/java/nl/nn/adapterframework/configuration/BasicAdapterServiceImpl.java

```diff
@@ -1,47 +1,77 @@
+/*
+   Copyright 2020 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
 package nl.nn.adapterframework.configuration;
 
-import nl.nn.adapterframework.core.IAdapter;
-import nl.nn.adapterframework.util.LogUtil;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.ObjectName;
 
 import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.jmx.export.MBeanExporter;
+
+import nl.nn.adapterframework.core.IAdapter;
+import nl.nn.adapterframework.util.LogUtil;
 
 /**
- * This implemention of {@link AdapterService} also registers the adapters to Jmx, and configures the registered Adapters.
+ * This implementation of {@link AdapterService} registers the adapters to a JMX server.
 
- * @author Michiel Meeuwissen
- * @since 5.0.29
+ * @author Niels Meijer
  */
-public class BasicAdapterServiceImpl extends AdapterServiceImpl {
-
-    private static final Logger LOG = LogUtil.getLogger(BasicAdapterServiceImpl.class);
-
-    @Override
-    public void registerAdapter(IAdapter adapter) throws ConfigurationException {
-        super.registerAdapter(adapter);
-        try {
-            // Throws javax.management.InstanceAlreadyExistsException when testing on
-            // WebSphere 7. This code has probably never been enabled as previously it was
-            // part of Configuration.java and was surrounded with "if (isEnableJMX())" with
-            // enableJMX being false by default.
-            LOG.debug("Registering adapter [" + adapter.getName() + "] to the JMX server");
-            JmxMbeanHelper.hookupAdapter(adapter);
-            LOG.info("[" + adapter.getName() + "] registered to the JMX server");
-        } catch (Throwable t) {
-            LOG.warn(t.getMessage());
-        }
-        adapter.configure();
-    }
-
-    @Override
-    public void unRegisterAdapter(IAdapter adapter) {
-        super.unRegisterAdapter(adapter);
-        try {
-            JmxMbeanHelper.unhookAdapter(adapter);
-        } catch (Throwable t) {
-            LOG.warn(t.getMessage());
-        }
-    }
+public class BasicAdapterServiceImpl extends AdapterServiceImpl implements ApplicationContextAware, InitializingBean {
```

Comment:
Laten we deze class JmxRegisteringAdapterService noemen, want dat is wat hij doet. Er is niks basics aan.